compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategiesGPU.cpp

```diff
@@ -63,6 +63,26 @@ auto unpackRegisteredMatchCallback(ImplicitLocOpBuilder &b,
   return std::tuple_cat(a);
 }
 
+/// Return max(1, (value * 32) / bitwidth).
+static int64_t scaleUpByBitWidth(int64_t value, int64_t bitWidth) {
+  assert((bitWidth & bitWidth - 1) == 0 && "bitWidth must be a power of 2");
+  return std::max((value * 32) / bitWidth, 1l);
+}
+
+/// Adjust the number of warps to use to benefit from packing multiple smaller
+/// elemental types within a single 128 bit shuffled element.
+static int64_t adjustNumberOfWarpsForBlockShuffle(int64_t numWarpsToUse,
+                                                  int64_t bitWidth) {
+  // Try to scale down the number of warps to use 32b elements in warp shuffles.
+  assert((bitWidth & bitWidth - 1) == 0 && "bitWidth must be a power of 2");
+  int64_t factor;
+  for (factor = std::max(32l / bitWidth, 1l); factor >= 1; factor >>= 1)
```

Comment:
Should the condition be `factor > 1`? I see that when we reach `factor == 1`, we will break out of the loop in all cases because `whatever % 1 == 0`, but this feels like a footgun if the break condition changes.

Reply:
Also nit: isn't the initialization condition `scaleUpByBitWidth(1, bitWidth)`?